grype/vex/openvex/implementation.go

```diff
@@ -0,0 +1,237 @@
+package openvex
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/anchore/grype/grype/match"
+	"github.com/anchore/grype/grype/pkg"
+	"github.com/anchore/packageurl-go"
+	"github.com/anchore/syft/syft/source"
+	openvex "github.com/openvex/go-vex/pkg/vex"
+	vexctl "github.com/openvex/vexctl/pkg/ctl"
+)
+
+type Processor struct{}
+
+func New() *Processor {
+	return &Processor{}
+}
+
+// ReadVexDocuments reads and merges VEX documents
+func (ovm *Processor) ReadVexDocuments(docs []string) (interface{}, error) {
+	// Combine all VEX documents into a single VEX document
+	vexdata, err := vexctl.New().MergeFiles(context.Background(), &vexctl.MergeOptions{}, docs)
+	if err != nil {
+		return nil, fmt.Errorf("merging vex documents: %w", err)
+	}
+
+	return vexdata, nil
+}
+
+// productIDentifiersFromContext reads the package context and returns software
+// identifiers identifying the scanned image.
+func productIDentifiersFromContext(pkgContext *pkg.Context) ([]string, error) {
+	switch v := pkgContext.Source.Metadata.(type) {
+	case source.StereoscopeImageSourceMetadata:
+		// TODO(puerco): We can create a wider definition here. This effectivley
+		// adds the multiarch image and the image of the OS running grype. We
+		// could generate more identifiers to match better.
+		return identifiersFromDigests(v.RepoDigests), nil
+	default:
+		// Fail for now
+		return nil, errors.New("source type not supported for VEX")
+	}
+}
+
+func identifiersFromDigests(digests []string) []string {
+	identifiers := []string{}
+
+	for _, d := range digests {
+		// The first identifier is the original image reference:
+		identifiers = append(identifiers, d)
+
+		// Now, parse the digest
+		parts := strings.SplitN(d, "@", 2)
```

Comment:
nit: this section could probably use either ggcr or docker/distribution libs to parse the ref and extract the digest:
```golang
                 value = ""
		ref, err := reference.Parse(d)
		if ref, ok := ref.(reference.Digested); ok {
			value = ref.Digest().String()
		}
```

Reply:
I've rewritten it to use GGCR (and added a test) :+1: